simulator/source/Combat_simulator.cpp

```diff
@@ -1535,8 +1557,9 @@ void Combat_simulator::simulate(const Character& character, int init_iteration,
                     else if (slam_manager.time_left(time_keeper_.time) <= 0.0)
                     {
                         slam(weapons[0], special_stats, rage, damage_sources, flurry_charges, rampage_stacks, rampage_active);
-                        slam_manager.un_queue_slam();
-                        weapons[0].internal_swing_timer = weapons[0].swing_speed / (1 + special_stats.haste);
+                        slam_manager.finish_slam();
+                        weapons[0].internal_swing_timer = 0;
+                        update_swing_timers(weapons[0], is_dual_wield ? weapons[1] : weapons[0], oldHaste, special_stats.haste);
```

Comment:
"start swinging again" - note that "oldHaste" would have to be updated here if the end-of-turn update_swing_timers() call would adjust any internal_swing_timers mid-swing, but this isn't the case here

note: using a dummy off-hand weapon (~with socket = Socket::none or the like) would make a ton of function calls alot simpler